saleor/graphql/core/scalars.py

```diff
@@ -46,18 +46,29 @@ def parse_value(value):
             return None
 
 
-class PositiveDecimal(Decimal):
+class PositiveDecimal(graphene.Float):
     """Nonnegative Decimal scalar implementation.
 
     Should be used in places where value must be nonnegative (0 or greater).
     """
 
     @staticmethod
-    def parse_value(value):
-        value = Decimal.parse_value(value)
-        if value and value < 0:
-            return None
-        return value
+    def parse_value(value) -> decimal.Decimal | None:
+        parsed_value = Decimal.parse_value(value)
+
+        if (parsed_value is not None) and parsed_value >= 0:
+            return parsed_value
+
+        return None
```

Comment:
I have simplified that:
- don't overwrite value, which is less bug-prone and mypy works (parsed_value is decimal, value is any)
- parsed_value can be 0 or None, which is bug-prone in "if" statements (we allow zero, don't allow null)